idb/postgres/postgres_migrations.go

```diff
@@ -1165,32 +1165,28 @@ func sqlMigration(db *IndexerDb, state *MigrationState, sqlLines []string) error
 	defer db.accountingLock.Unlock()
 
 	thisMigration := state.NextMigration
-	tx, err := db.db.BeginTx(context.Background(), &serializable)
-	if err != nil {
-		db.log.WithError(err).Errorf("migration %d tx err", thisMigration)
-		return err
-	}
-	defer tx.Rollback() // ignored if .Commit() first
-	for i, cmd := range sqlLines {
-		_, err = tx.Exec(cmd)
+	state.NextMigration++
```

Comment:
Because `state` is passed by reference, please wait to update it until after the migration is successful

Reply:
`upsertMigrationStateTx()` updates it before talking to DB. I don't think there is a good solution, especially here because `f()` can be called multiple time.

Reply:
```go
nextState := *state
nextState.NextMigration++
...
*state = nextState
```